File: components/FlagIcon.js
import styles from '../styles/langpicker.module.css';

/**
 * This function uses the langpicker css style to determine the language to use from the countryCode given.
 * It displays a circle with the flag of said country
 * @param {String} countryCode The code of a country to determine its language in two characters format
 * @returns A span tag representing a circle with the flag of the given countrycode
 */
export default function FlagIcon({ countryCode = "" }) {
    // Determines if we want the us or gb flag for the english language
    if (countryCode === "en") {
      countryCode = "gb"
    }
  
    return (
      <span className={`fi fis ${styles.fiCircle} inline-block mr-2 fi-${countryCode}`} />
    )
  }

File: src/components/flag_icon.rs
use leptos::*;


/// A small helper component to display a flag icon based on the country code.
/// The country code is expected to be a two-letter ISO 3166-1 alpha-2 code.
/// If the country code is "en", it will be replaced with "gb" to display the UK flag.
#[component]
pub fn FlagIcon(
    country_code: String,
) -> impl IntoView {
    let mut flag_icon = country_code.clone();
    flag_icon = flag_icon.to_lowercase();
    if flag_icon == "en" {
        flag_icon = "gb".to_string();
    }

    view! {
        <span class=format!("fi fis fiCircle inline-block mr-2 fi-{}", flag_icon) />
    }
}
